### PR TITLE
workaround for composition not ending in firefox on linux

### PR DESCRIPTION
**Pull request by nightwing**
fixes https://github.com/replit/codemirror-vim/issues/98, but needs more testing on mac

demo: https://raw.githack.com/replit/codemirror-vim/linux-composition/dev/web-demo.html

### CODE REVIEW

**Comment by sergeichestakov** [comment]
do we even need this on Mac? can we just gate this to Linux by checking the user agent?

**Comment by nightwing** [comment]
We don't exactly need this, but it would have been nice to not rely on useragent when deciding this behavior.
Unfortunately the simple solution did not work on mac, switching to french keyboard, pressing `[` to input `^` then pressing `i` moves cursor to the start of line but doesn't enter insert mode because composition stays active and the key `î` is inserted instead of `i`.
This new version should work both on mac and on firefox linux.

**Comment by sergeichestakov** [comment]
yeah I'm not super concerned about that since there's a lot of platform/user-agent checks upstream, especially in CM View, due to subtle differences/bugs on each platform and browser

Wait so this will work on Mac then too? is the user agent check not necessary then?

**Comment by nightwing** [comment]
This will work on mac because it has user agent check.

File: src/index.ts [hunk 1]
```diff
@@ -296,6 +300,9 @@ const vimPlugin = ViewPlugin.fromClass(
           var vimPlugin = cm.state.vimPlugin;
 
           if (vim && !vim.insertMode && !cm.curOp?.isVimOp) {
+            if (text == "\0\0") {
```
Comment (sergeichestakov) [inline src/index.ts:303]: we should use triple equals here
```suggestion
            if (text === "\0\0") {
```

**Review by sergeichestakov — APPROVED** [review]
great, thanks for the fix! 🦊

**Pull request merged**